db/src/main/kotlin/edu/artic/db/AppDataManager.kt

```diff
@@ -39,7 +50,49 @@ class AppDataManager @Inject constructor(
 
 
     /**
-     * @return an observable that will show how close to completion we are and a Done if fully complete
+     * In-memory cache of the [ArticDataObject] currently backing
+     * this app.
+     *
+     * Use it to access important network communication config like
+     * [ArticDataObject.imageServerUrl] and [ArticDataObject.dataApiUrl].
+     *
+     * Values are sourced from [the ArticAppData][ArticAppData.data]
+     * parsed in [getBlob]. Be advised that if no such value has been
+     * yet found in the time since first install, this will provide only
```

Comment:
no nulls allowed in RXJava, so this will _not_ emit.

Reply:
so either we make this `Optional<ArticDataObject>` or we're ok with registered listeners not emitting.

Reply:
Hmm. I think we'd be ok with it not emitting. What I would like you to advise on is if we could somehow inject this data into the sundry DAOs so that the object is available to the `asCDNUri` function. I don't see how this could fail to emit in such a case.

Reply:
right, just a warning. any chain you expect to continue or depend on this object will not emit a value _until_ something is put into the DB, which should be fine.